client-api/src/main/java/org/sonarsource/sonarlint/core/clientapi/backend/binding/BindingSuggestionService.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * SonarLint Core - Client API
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.clientapi.backend.binding;
+
+import java.util.concurrent.CompletableFuture;
+import org.eclipse.lsp4j.jsonrpc.services.JsonRequest;
+import org.sonarsource.sonarlint.core.clientapi.client.SuggestBindingResponse;
+
+public interface BindingSuggestionService {
+
+  /**
+   * Calculates a suggested binding for a 'configScopeId' and 'connectionId' specified in the {@link GetBindingSuggestionParams}
+   * @return {@link SuggestBindingResponse} containing binding suggestions
+   */
+  @JsonRequest
+  CompletableFuture<SuggestBindingResponse> suggestBinding(GetBindingSuggestionParams params);
```

Comment:
There is a risk of confusion with the notification on the client interface. Maybe we should find a different name?
```suggestion
  CompletableFuture<GetBindingSuggestionsResponse> getBindingSuggestions(GetBindingSuggestionsParams params);
```

Reply:
Good point; Addressed in [078e014](https://github.com/SonarSource/sonarlint-core/pull/594/commits/078e01419b054088432516cdda1ff62c8418bd96)